src/core/simulation_results.jl

```diff
@@ -1,517 +1,378 @@
-struct SimulationResultsReference
-    ref::Dict
-    results_folder::String
-    chronologies::Dict
-    base_powers::Dict
+const FILE_STRUCT =
+    ["data_store", "logs", "models_json", "recorder", "results", "simulation_files"]
+
+mutable struct SimulationResults <: PSIResults
+    stage::String
+    base_power::Float64
+    execution_path::AbstractString
+    results_output_folder::String
+    existing_timestamps::StepRange{Dates.DateTime, Dates.Millisecond}
+    results_timestamps::Vector{Dates.DateTime}
+    system::Union{Nothing, PSY.System}
+    existing_variables::Array{Symbol}
```

Comment:
Would it be possible to remove these `existing_` members by instead storing them inside the `_values` members? At initial construction you could add each field to the outer dict. The inner dict would remain empty until the user performs a read.